src/Microsoft.IdentityModel.Tokens/CompressionProviderFactory.cs

```diff
@@ -0,0 +1,119 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using Microsoft.IdentityModel.Logging;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// Compression provider factory for compression and decompression.
+    /// </summary>
+    public class CompressionProviderFactory
+    {
+        private static CompressionProviderFactory _default;
+
+        /// <summary>
+        /// Static constructor that initializes the default <see cref="CompressionProviderFactory"/>.
+        /// </summary>
+        static CompressionProviderFactory()
+        {
+            Default = new CompressionProviderFactory();
+        }
+
+        /// <summary>
+        /// Default constructor for <see cref="CompressionProviderFactory"/>.
+        /// </summary>
+        public CompressionProviderFactory()
+        {
+        }
+
+        /// <summary>
+        /// Constructor that creates a deep copy of given <see cref="CompressionProviderFactory"/> object.
+        /// </summary>
+        /// <param name="other"><see cref="CompressionProviderFactory"/> to copy from.</param>
+        public CompressionProviderFactory(CompressionProviderFactory other)
+        {
+            if (other == null)
+                throw LogHelper.LogArgumentNullException(nameof(other));
+
+            CustomCompressionProvider = other.CustomCompressionProvider;
+        }
+
+        /// <summary>
+        /// Returns the default <see cref="CompressionProviderFactory"/> instance.
+        /// </summary>
+        public static CompressionProviderFactory Default
+        {
+            get => _default;
+            set => _default = value ?? throw LogHelper.LogArgumentNullException(nameof(Default));
+        }
+
+        /// <summary>
+        /// Extensibility point for custom compression support application wide.
+        /// </summary>
+        public CompressionProvider CustomCompressionProvider { get; set; }
+
+        /// <summary>
+        /// Answers if an algorithm is supported.
+        /// </summary>
+        /// <param name="algorithm">the name of the crypto algorithm.</param>
+        /// <returns>true if the algorithm is supported, false otherwise.</returns>
+        public virtual bool IsSupportedAlgorithm(string algorithm)
+        {
+            if (CustomCompressionProvider != null && CustomCompressionProvider.IsSupportedAlgorithm(algorithm))
+                return true;
+
+            return IsSupportedCompressionAlgorithm(algorithm);
+        }
+
+        private bool IsSupportedCompressionAlgorithm(string algorithm)
+        {
+            return CompressionAlgorithms.Deflate.Equals(algorithm);
```

Comment:
It would be nice to specify a string comparison algorithm here. For example, StringComparison.Ordinal or StringComparison.OrdinalIgnoreCase depending on whether or not you care about case sensitivity. If you don't, Ordinal has significantly better performance.

Reply:
I've changed this line to use StringComparison.Ordinal.